src/components/Card/CardList.tsx

```diff
@@ -7,9 +7,25 @@ import { mockCards } from "../../services/mockCardData";
 import { logger } from "../../utils/logger";
 import { CardItem } from "./CardItem";
 
-export function CardList() {
+export interface CardListProps {
+  category: string;
+}
+
+export function CardList({ category }: CardListProps) {
```

Comment:
Better to use `react-router-dom` hooks instead, since now every parent component of `CardList` needs to know what category is being passed in, and also means that the `fetch` or `loader` component cannot be used in another tool. A hook also adds the benefit of controlling renders more finely, because we can render the component without knowledge of what is inside (it will render an empty/placeholder `<CardList>` while loading the fetch, then once it is fetched the component will be rendered). Doing it this way means there isn't a `loading` process.

See: https://reactrouter.com/en/main/hooks/use-params

Example refactor:
```typescript
import { Routes, Route, useParams } from 'react-router-dom';

export function CardList() {
  const { category } = useParams();
  const cards = category === ""
      ? mockCards
      : mockCards.filter((mockCard) => mockCard.categories.includes(category));
  ...
}
```

Reply:
Thanks! I was clearly confused about where I could use `useParams`, since I'd used it in `Main.tsx` 😅 Updated!